pkg/lifecycle/once.go

```diff
@@ -85,20 +75,20 @@ type lifecycleOnce struct {
 	// err is conferred to whichever goroutine is starting or stopping, until
 	// it has started or stopped, after which `err` becomes immutable.
 	err syncatomic.Value
-	// state is an atomic LifecycleState representing the object's current
+	// state is an atomic State representing the object's current
 	// state (Idle, Starting, Running, Stopping, Stopped, Errored).
 	state atomic.Int32
 }
 
-// Once returns a lifecycle controller.
+// NewOnce returns a lifecycle controller.
 // 0. The observable lifecycle state must only go forward from birth to death.
 // 1. Start() must block until the state is >= Running
 // 2. Stop() must block until the state is >= Stopped
 // 3. Stop() must pre-empt Start() if it occurs first
 // 4. Start() and Stop() may be backed by a do actual work function, and that
 //    function must be called at most once.
-func Once() LifecycleOnce {
-	return &lifecycleOnce{
+func NewOnce() *Once {
```

Comment:
I know this hasn't changed but since this is going to show up in public godocs:

Godocs doesn't render itemized lists. This will render as a massive unreadable paragraph. Either convert to prose or indent itemized list so that it renders as a code block.